awss3/store.go

```diff
@@ -0,0 +1,703 @@
+package awss3
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"math/rand"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	u "github.com/araddon/gou"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/pborman/uuid"
+	"golang.org/x/net/context"
+
+	"github.com/lytics/cloudstorage"
+)
+
+const (
+	// StoreType = "s3" this is used to define the storage type to create
+	// from cloudstorage.NewStore(config)
+	StoreType = "s3"
+
+	// Configuration Keys.  These are the names of keys
+	// to look for in the json map[string]string to extract for config.
+
+	// ConfKeyAccessKey config key name of the aws access_key(id) for auth
+	ConfKeyAccessKey = "access_key"
+	// ConfKeyAccessSecret config key name of the aws acccess secret
+	ConfKeyAccessSecret = "access_secret"
+	// ConfKeyARN config key name of the aws ARN name of user
+	ConfKeyARN = "arn"
+	// ConfKeyDisableSSL config key name of disabling ssl flag
+	ConfKeyDisableSSL = "disable_ssl"
+	// Authentication Source's
+
+	// AuthAccessKey is for using aws access key/secret pairs
+	AuthAccessKey cloudstorage.AuthMethod = "aws_access_key"
+)
+
+var (
+	// Retries number of times to retry upon failures.
+	Retries = 55
```

Comment:
We could consider lowering this, I believe I set it so high for GCS because in the old days we saw lots of errors for long periods of time (~10mins).  That's not the case anymore.

Reply:
done